src/wildcard.cpp

```diff
@@ -328,6 +328,67 @@ bool wildcard_match(const wcstring &str, const wcstring &wc, bool leading_dots_f
     return match != fuzzy_match_none;
 }
 
+static int fast_waccess(const struct stat &stat_buf, uint8_t mode) {
+    // Cache the effective user id and group id of our own shell process. These can't change on us
+    // because we don't change them.
+    static const uid_t euid = geteuid();
+    static const gid_t egid = getegid();
+
+    // Cache a list of our group memberships.
+    static const std::vector<gid_t> groups = ([&]() {
+        std::vector<gid_t> groups;
+        while (true) {
+            int ngroups = getgroups(0, nullptr);
+            assert(ngroups > 0 && "getgroups(0, nullptr) returned 0 group memberships!");
```

Comment:
I think technically you can have no supplementary groups? The spec is not very clear.

Reply:
I forgot that the effective group/primary group is not guaranteed to be included and was presuming that would be the at least one response. You're right, we should probably remove that hard assert.